source/common/stats/metric_impl.cc

```diff
@@ -15,20 +15,20 @@ MetricHelper::~MetricHelper() {
 }
 
 MetricHelper::MetricHelper(StatName name, absl::string_view tag_extracted_name,
-                           const std::vector<Tag>& tags, SymbolTable& symbol_table) {
+                           const StatNameTagVector& stat_name_tags, SymbolTable& symbol_table) {
   // Encode all the names and tags into transient storage so we can count the
   // required bytes. 2 is added to account for the name and tag_extracted_name,
   // and we multiply the number of tags by 2 to account for the name and value
   // of each tag.
-  const uint32_t num_names = 2 + 2 * tags.size();
+  const uint32_t num_names = 2 + 2 * stat_name_tags.size();
   absl::FixedArray<StatName> names(num_names);
   names[0] = name;
   StatNamePool pool(symbol_table);
```

Comment:
optional trivial perf nit. Now that you have removed the other uses of 'pool' here it would be slightly more efficient to write these 2 lines as:
```
  StatNameManagedStorage storage(tag_extracted_name, symbol_table);
  names[1] = storage.statName();
```
This will avoid creating/destroying the slightly more complex `StatNamePool` object just to allocate a single statName.